valohai/metadata.py

```diff
@@ -73,8 +78,9 @@ def flush(self) -> None:
         This will log all three metrics at once.
         """
         if self.partial_logs:
-            # Start with \n, ensuring JSON prints on it's own line
-            print(f"\n{json.dumps(self.partial_logs, default=str)}")  # noqa
+            # Start with \n, ensuring JSON prints on its own line
+            to_print = {'vh_metadata': self.partial_logs} if is_valohai_deployment() else self.partial_logs
+            print(f"\n{json.dumps(to_print, default=str)}")
```

Comment:
Might be clearer as something like
```suggestion
            to_print = self.partial_logs
            if is_valohai_deployment():
                # Wrap in `vh_metadata` so deployment log machinery detects this
                to_print = {"vh_metadata": to_print}
            # Start with \n, ensuring JSON prints on its own line
            print(f"\n{json.dumps(to_print, default=str)}")
```